src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/TdsParserStateObject.cs

```diff
@@ -1985,12 +2032,13 @@ internal bool TryReadNetworkPacket()
 
                 // previous buffer is in snapshot
                 _inBuff = new byte[_inBuff.Length];
+                result = TdsOperationStatus.NeedMoreData;
```

Comment:
Is this unintentional? `result` is being assigned  `NeedMoreData` which will be treated as `!= Done` by the callers of this function, which would lead to an exception, which shouldn't be the case.

Reply:
I re-read code. It looks like we return `false` when we have the previous buffer in snapshot. 
Hence your change of setting result to `TdsOperationStatus.NeedMoreData` should not cause any impact from what the code is doing right now. This could be fine. 
My previous comment about "an exception being thrown" is incorrect. 

However for the spirit of this PR, can you remove this line, "2035" or comment it out, so that this looks similar to what the current code is doing, except replace of bool with tri-state enum?

Reply:
Removed.